x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportPreviewTransformAction.java

```diff
@@ -170,6 +174,8 @@ private void getPreview(
                 Clock.systemUTC()
             );
 
+            List<String> warnings = TransformWarnings.getWarnings(function, source, syncConfig);
+            warnings.forEach(HeaderWarning::addWarning);
```

Comment:
how does the UI react on this?

Reply:
Currently the Transform UI does not do anything with these warnings (most importantly, it doesn't fail).
Having said that, from what I can see the `_preview` request to the API does not contain `sync` part and without `sync` no warnings are generated. I'll need to consult UI team if that's intended.

In Dev Tools, OTOH, the warnings are displayed with the "Deprecation" prefix:
```
#! Deprecation: all the group-by fields are script-based runtime fields, this transform might run slowly, please check your configuration.
#! Deprecation: sync time field is a script-based runtime field, this transform might run slowly, please check your configuration.
```

So I'll need to double-check whether
1. there is anything in the returned header that makes the UI think it's about deprecation
2. or Dev Tools renders *all* header warnings as deprecations.

Reply:
I consulted ML UI team and currently it's expected that the `_preview` request doesn't include `sync` section. So my is no-op as far as UI is concerned.
I looked into the Dev Tools code and it seems *all* the header warnings are rendered with "Deprecation" prefix. I haven't got an answer yet whether this is intentional or it just happened that until now there were no non-deprecation warnings.